data/kata-collect-data.sh.in

```diff
@@ -0,0 +1,561 @@
+EOT
+}
+
+die()
+{
+	local msg="$*"
+	echo >&2 "ERROR: $script_name: $msg"
+	exit 1
+}
+
+msg()
+{
+	local msg="$*"
+	echo "$msg"
+}
+
+heading()
+{
+	local name="$*"
+	echo -e "\n# $name\n"
+}
+
+subheading()
+{
+	local name="$*"
+	echo -e "\n## $name\n"
+}
+
+separator()
+{
+	echo -e '\n---\n'
+}
+
+have_cmd()
+{
+	local cmd="$1"
+
+	command -v "$cmd" &>/dev/null
+	local ret=$?
+
+	if [ $ret -eq 0 ]; then
+		msg "Have \`$cmd\`"
+	else
+		msg "No \`$cmd\`"
+	fi
+
+	[ $ret -eq 0 ]
+}
+
+show_quoted_text()
+{
+	local text="$*"
+
+	echo "\`\`\`"
+	echo "$text"
+	echo "\`\`\`"
+}
+
+run_cmd_and_show_quoted_output()
+{
+	local cmd="$*"
+
+	local output
+
+	msg "Output of \"\`$cmd\`\":"
+	output=$(eval "$cmd" 2>&1)
+	show_quoted_text "$output"
+}
+
+show_runtime_configs()
+{
+	local configs config
+
+	heading "Runtime config files"
+	
+	configs=$($runtime --@PROJECT_TYPE@-show-default-config-paths)
+	if [ $? -ne 0 ]; then
+		version=$($runtime --version|tr '\n' ' ')
+		die "failed to check config files - runtime is probably too old ($version)"
+	fi
+
+	subheading "Runtime default config files"
+
+	show_quoted_text "$configs"
+
+	# add in the standard defaults for good measure "just in case"
+	configs+=" /etc/@PROJECT_TAG@/configuration.toml"
+	configs+=" /usr/share/defaults/@PROJECT_TAG@/configuration.toml"
+	configs+=" @DESTCONFIG@"
+	configs+=" @DESTSYSCONFIG@"
+
+	# create a unique list of config files
+	configs=$(echo $configs|tr ' ' '\n'|sort -u)
+
+	subheading "Runtime config file contents"
+
+	for config in $configs; do
+		if [ -e "$config" ]; then
+			run_cmd_and_show_quoted_output "cat \"$config\""
+		else
+			msg "Config file \`$config\` not found"
+		fi
+	done
+
+	separator
+}
+
+show_log_details()
+{
+	heading "Logfiles"
+
+	show_runtime_log_details
+	show_proxy_log_details
+	show_shim_log_details
+
+	separator
+}
+
+show_runtime_log_details()
+{
+	subheading "Runtime logs"
+
+	find_system_journal_problems "runtime" "@RUNTIME_NAME@" ""
+}
+
+find_system_journal_problems()
+{
+	local name="$1"
+	local program="$2"
+	local unit="$3"
+
+	# select by identifier
+	local selector="-t"
+
+	local data_source="system journal"
+
+	local problems=$(journalctl -q -o cat -a "$selector" "$program" |\
+		grep "time=" |\
+		egrep -i "$problem_pattern" |\
+		egrep -iv "$problem_exclude_pattern" |\
+		tail -n ${PROBLEM_LIMIT})
+
+	if [ -n "$problems" ]; then
+		msg "Recent $name problems found in $data_source:"
+		show_quoted_text "$problems"
+	else
+		msg "No recent $name problems found in $data_source."
+	fi
+}
+
+show_proxy_log_details()
+{
+	subheading "Proxy logs"
+
+	find_system_journal_problems "proxy" "@PROJECT_TYPE@-proxy" ""
+}
+
+show_shim_log_details()
+{
+	subheading "Shim logs"
+
+	find_system_journal_problems "shim" "@PROJECT_TYPE@-shim" ""
+}
+
+show_package_versions()
+{
+	heading "Packages"
+
+	local pattern="("
+	local project
+
+	# CC 2.x runtime. This shouldn't be installed but let's check anyway
+	pattern+="cc-oci-runtime"
```

Comment:
here we are considering to cc-oci-runtime (2.x), I wonder if cc-runtime (3.0) should be consider too